CacheSim/CacheSim.h

```diff
@@ -28,25 +28,63 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include <stdint.h>
 #include <stdio.h>
+#if defined(_MSC_VER)
 #include <Windows.h>
+#ifndef IG_CACHESIM_API
+#define IG_CACHESIM_API __declspec(dllimport)
+#endif
+#else
+#include <dlfcn.h>
+#include <unistd.h>
+#define IG_CACHESIM_API
```

Comment:
Sure, I'll look into that. I don't have a ton of experience with building solibs. The idea here is to make it so that only the IG_CACHESIM_API calls are visible symbols during loading?